isb_lib/opencontext_adapter.py

```diff
@@ -148,6 +149,21 @@ def _getPage(self):
     def last_url_str(self) -> typing.AnyStr:
         return self.url
 
+def _validate_resolved_content(thing: igsn_lib.models.thing.Thing):
+    if not isinstance(thing.resolved_content, dict):
+        raise ValueError("Thing.resolved_content is not an object")
+    if not thing.authority_id == OpenContextItem.AUTHORITY_ID:
+        raise ValueError("Thing is not an OpenContext item")
+
+def reparse_as_core_record(thing: igsn_lib.models.thing.Thing) -> typing.Dict:
+    _validate_resolved_content(thing)
+    try:
+        transformer = isamples_metadata.OpenContextTransformer.OpenContextTransformer(thing.resolved_content)
+        return isb_lib.core.coreRecordAsSolrDoc(transformer.transform())
+    except Exception as e:
+        get_logger().fatal("Failed trying to run transformer on %s", str(thing.resolved_content))
```

Comment:
The lat/lon False values was one major problem we encountered, and there was one more.  It turns out a number of the OpenContext records didn't have persistent identifiers, which caused the transformer to blow up.  Per discussion with @ekansa, this is not an expected valid scenario so our assumption that all records would have a persistent is correct.  In the meantime, we log the exception and skip the record during import.  Once it's fixed, we will need to fetch those records and rerun the solr importer.  It ends up being ~2200 records out of > 800k, so most are ok.